File: java-play-framework/app/apimodels/MessagesBase.java
package apimodels;

import apimodels.EmojiReaction;
import apimodels.MessagesBaseTopicLinks;
import apimodels.OneOfstringarray;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * Object containing details of the message. 
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen")
@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class MessagesBase   {
  @JsonProperty("avatar_url")
  
  private String avatarUrl;

  @JsonProperty("client")
  
  private String client;

  @JsonProperty("content")
  
  private String content;

  @JsonProperty("content_type")
  
  private String contentType;

  @JsonProperty("display_recipient")
  @Valid

  private OneOfstringarray displayRecipient;

  @JsonProperty("id")
  
  private Integer id;

  @JsonProperty("is_me_message")
  
  private Boolean isMeMessage;

  @JsonProperty("reactions")
  @Valid

  private List<EmojiReaction> reactions = null;

  @JsonProperty("recipient_id")
  
  private Integer recipientId;

  @JsonProperty("sender_email")
  
  private String senderEmail;

  @JsonProperty("sender_full_name")
  
  private String senderFullName;

  @JsonProperty("sender_id")
  
  private Integer senderId;

  @JsonProperty("sender_realm_str")
  
  private String senderRealmStr;

  @JsonProperty("stream_id")
  
  private Integer streamId;

  @JsonProperty("subject")
  
  private String subject;

  @JsonProperty("topic_links")
  @Valid

  private List<MessagesBaseTopicLinks> topicLinks = null;

  @JsonProperty("submessages")
  
  private List<String> submessages = null;

  @JsonProperty("timestamp")
  
  private Integer timestamp;

  @JsonProperty("type")
  
  private String type;

  public MessagesBase avatarUrl(String avatarUrl) {
    this.avatarUrl = avatarUrl;
    return this;
  }

   /**
   * The URL of the user's avatar.  Can be null only if client_gravatar was passed, which means that the user has not uploaded an avatar in Zulip, and the client should compute the gravatar URL by hashing the user's email address itself for this user. 
   * @return avatarUrl
  **/
  public String getAvatarUrl() {
    return avatarUrl;
  }

  public void setAvatarUrl(String avatarUrl) {
    this.avatarUrl = avatarUrl;
  }

  public MessagesBase client(String client) {
    this.client = client;
    return this;
  }

   /**
   * A Zulip \"client\" string, describing what Zulip client sent the message. 
   * @return client
  **/
  public String getClient() {
    return client;
  }

  public void setClient(String client) {
    this.client = client;
  }

  public MessagesBase content(String content) {
    this.content = content;
    return this;
  }

   /**
   * The content/body of the message. 
   * @return content
  **/
  public String getContent() {
    return content;
  }

  public void setContent(String content) {
    this.content = content;
  }

  public MessagesBase contentType(String contentType) {
    this.contentType = contentType;
    return this;
  }

   /**
   * The HTTP `content_type` for the message content.  This will be `text/html` or `text/x-markdown`, depending on whether `apply_markdown` was set. 
   * @return contentType
  **/
  public String getContentType() {
    return contentType;
  }

  public void setContentType(String contentType) {
    this.contentType = contentType;
  }

  public MessagesBase displayRecipient(OneOfstringarray displayRecipient) {
    this.displayRecipient = displayRecipient;
    return this;
  }

   /**
   * Data on the recipient of the message; either the name of a stream or a dictionary containing basic data on the users who received the message. 
   * @return displayRecipient
  **/
  public OneOfstringarray getDisplayRecipient() {
    return displayRecipient;
  }

  public void setDisplayRecipient(OneOfstringarray displayRecipient) {
    this.displayRecipient = displayRecipient;
  }

  public MessagesBase id(Integer id) {
    this.id = id;
    return this;
  }

   /**
   * The unique message ID.  Messages should always be displayed sorted by ID. 
   * @return id
  **/
  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public MessagesBase isMeMessage(Boolean isMeMessage) {
    this.isMeMessage = isMeMessage;
    return this;
  }

   /**
   * Whether the message is a [/me status message][status-messages]  [status-messages]: /help/format-your-message-using-markdown#status-messages 
   * @return isMeMessage
  **/
  public Boolean getIsMeMessage() {
    return isMeMessage;
  }

  public void setIsMeMessage(Boolean isMeMessage) {
    this.isMeMessage = isMeMessage;
  }

  public MessagesBase reactions(List<EmojiReaction> reactions) {
    this.reactions = reactions;
    return this;
  }

  public MessagesBase addReactionsItem(EmojiReaction reactionsItem) {
    if (reactions == null) {
      reactions = new ArrayList<>();
    }
    reactions.add(reactionsItem);
    return this;
  }

   /**
   * Data on any reactions to the message. 
   * @return reactions
  **/
  public List<EmojiReaction> getReactions() {
    return reactions;
  }

  public void setReactions(List<EmojiReaction> reactions) {
    this.reactions = reactions;
  }

  public MessagesBase recipientId(Integer recipientId) {
    this.recipientId = recipientId;
    return this;
  }

   /**
   * A unique ID for the set of users receiving the message (either a stream or group of users).  Useful primarily for hashing. 
   * @return recipientId
  **/
  public Integer getRecipientId() {
    return recipientId;
  }

  public void setRecipientId(Integer recipientId) {
    this.recipientId = recipientId;
  }

  public MessagesBase senderEmail(String senderEmail) {
    this.senderEmail = senderEmail;
    return this;
  }

   /**
   * The Zulip display email address of the message's sender. 
   * @return senderEmail
  **/
  public String getSenderEmail() {
    return senderEmail;
  }

  public void setSenderEmail(String senderEmail) {
    this.senderEmail = senderEmail;
  }

  public MessagesBase senderFullName(String senderFullName) {
    this.senderFullName = senderFullName;
    return this;
  }

   /**
   * The full name of the message's sender. 
   * @return senderFullName
  **/
  public String getSenderFullName() {
    return senderFullName;
  }

  public void setSenderFullName(String senderFullName) {
    this.senderFullName = senderFullName;
  }

  public MessagesBase senderId(Integer senderId) {
    this.senderId = senderId;
    return this;
  }

   /**
   * The user ID of the message's sender. 
   * @return senderId
  **/
  public Integer getSenderId() {
    return senderId;
  }

  public void setSenderId(Integer senderId) {
    this.senderId = senderId;
  }

  public MessagesBase senderRealmStr(String senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
    return this;
  }

   /**
   * A string identifier for the realm the sender is in.  Unique only within the context of a given Zulip server.  E.g. on `example.zulip.com`, this will be `example`. 
   * @return senderRealmStr
  **/
  public String getSenderRealmStr() {
    return senderRealmStr;
  }

  public void setSenderRealmStr(String senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
  }

  public MessagesBase streamId(Integer streamId) {
    this.streamId = streamId;
    return this;
  }

   /**
   * Only present for stream messages; the ID of the stream. 
   * @return streamId
  **/
  public Integer getStreamId() {
    return streamId;
  }

  public void setStreamId(Integer streamId) {
    this.streamId = streamId;
  }

  public MessagesBase subject(String subject) {
    this.subject = subject;
    return this;
  }

   /**
   * The `topic` of the message.  Currently always `\"\"` for private messages, though this could change if Zulip adds support for topics in private message conversations.  The field name is a legacy holdover from when topics were called \"subjects\" and will eventually change. 
   * @return subject
  **/
  public String getSubject() {
    return subject;
  }

  public void setSubject(String subject) {
    this.subject = subject;
  }

  public MessagesBase topicLinks(List<MessagesBaseTopicLinks> topicLinks) {
    this.topicLinks = topicLinks;
    return this;
  }

  public MessagesBase addTopicLinksItem(MessagesBaseTopicLinks topicLinksItem) {
    if (topicLinks == null) {
      topicLinks = new ArrayList<>();
    }
    topicLinks.add(topicLinksItem);
    return this;
  }

   /**
   * Data on any links to be included in the `topic` line (these are generated by [custom linkification filters](/help/add-a-custom-linkifier) that match content in the message's topic.)  **Changes**: This field contained a list of urls before   Zulip 4.0 (feature level 46).  New in Zulip 3.0 (feature level 1): Previously, this field was called `subject_links`; clients are recommended to rename `subject_links` to `topic_links` if present for compatibility with older Zulip servers. 
   * @return topicLinks
  **/
  public List<MessagesBaseTopicLinks> getTopicLinks() {
    return topicLinks;
  }

  public void setTopicLinks(List<MessagesBaseTopicLinks> topicLinks) {
    this.topicLinks = topicLinks;
  }

  public MessagesBase submessages(List<String> submessages) {
    this.submessages = submessages;
    return this;
  }

  public MessagesBase addSubmessagesItem(String submessagesItem) {
    if (submessages == null) {
      submessages = new ArrayList<>();
    }
    submessages.add(submessagesItem);
    return this;
  }

   /**
   * Data used for certain experimental Zulip integrations. 
   * @return submessages
  **/
  public List<String> getSubmessages() {
    return submessages;
  }

  public void setSubmessages(List<String> submessages) {
    this.submessages = submessages;
  }

  public MessagesBase timestamp(Integer timestamp) {
    this.timestamp = timestamp;
    return this;
  }

   /**
   * The UNIX timestamp for when the message was sent, in UTC seconds. 
   * @return timestamp
  **/
  public Integer getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(Integer timestamp) {
    this.timestamp = timestamp;
  }

  public MessagesBase type(String type) {
    this.type = type;
    return this;
  }

   /**
   * The type of the message: `stream` or `private`. 
   * @return type
  **/
  public String getType() {
    return type;
  }

  public void setType(String type) {
    this.type = type;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MessagesBase messagesBase = (MessagesBase) o;
    return Objects.equals(avatarUrl, messagesBase.avatarUrl) &&
        Objects.equals(client, messagesBase.client) &&
        Objects.equals(content, messagesBase.content) &&
        Objects.equals(contentType, messagesBase.contentType) &&
        Objects.equals(displayRecipient, messagesBase.displayRecipient) &&
        Objects.equals(id, messagesBase.id) &&
        Objects.equals(isMeMessage, messagesBase.isMeMessage) &&
        Objects.equals(reactions, messagesBase.reactions) &&
        Objects.equals(recipientId, messagesBase.recipientId) &&
        Objects.equals(senderEmail, messagesBase.senderEmail) &&
        Objects.equals(senderFullName, messagesBase.senderFullName) &&
        Objects.equals(senderId, messagesBase.senderId) &&
        Objects.equals(senderRealmStr, messagesBase.senderRealmStr) &&
        Objects.equals(streamId, messagesBase.streamId) &&
        Objects.equals(subject, messagesBase.subject) &&
        Objects.equals(topicLinks, messagesBase.topicLinks) &&
        Objects.equals(submessages, messagesBase.submessages) &&
        Objects.equals(timestamp, messagesBase.timestamp) &&
        Objects.equals(type, messagesBase.type);
  }

  @Override
  public int hashCode() {
    return Objects.hash(avatarUrl, client, content, contentType, displayRecipient, id, isMeMessage, reactions, recipientId, senderEmail, senderFullName, senderId, senderRealmStr, streamId, subject, topicLinks, submessages, timestamp, type);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MessagesBase {\n");
    
    sb.append("    avatarUrl: ").append(toIndentedString(avatarUrl)).append("\n");
    sb.append("    client: ").append(toIndentedString(client)).append("\n");
    sb.append("    content: ").append(toIndentedString(content)).append("\n");
    sb.append("    contentType: ").append(toIndentedString(contentType)).append("\n");
    sb.append("    displayRecipient: ").append(toIndentedString(displayRecipient)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    isMeMessage: ").append(toIndentedString(isMeMessage)).append("\n");
    sb.append("    reactions: ").append(toIndentedString(reactions)).append("\n");
    sb.append("    recipientId: ").append(toIndentedString(recipientId)).append("\n");
    sb.append("    senderEmail: ").append(toIndentedString(senderEmail)).append("\n");
    sb.append("    senderFullName: ").append(toIndentedString(senderFullName)).append("\n");
    sb.append("    senderId: ").append(toIndentedString(senderId)).append("\n");
    sb.append("    senderRealmStr: ").append(toIndentedString(senderRealmStr)).append("\n");
    sb.append("    streamId: ").append(toIndentedString(streamId)).append("\n");
    sb.append("    subject: ").append(toIndentedString(subject)).append("\n");
    sb.append("    topicLinks: ").append(toIndentedString(topicLinks)).append("\n");
    sb.append("    submessages: ").append(toIndentedString(submessages)).append("\n");
    sb.append("    timestamp: ").append(toIndentedString(timestamp)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: csharp-nancyfx/src/Org.OpenAPITools/Models/MessagesBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools._api_v1.Models
{
    /// <summary>
    /// Object containing details of the message. 
    /// </summary>
    public sealed class MessagesBase:  IEquatable<MessagesBase>
    { 
        /// <summary>
        /// The URL of the user&#39;s avatar.  Can be null only if client_gravatar was passed, which means that the user has not uploaded an avatar in Zulip, and the client should compute the gravatar URL by hashing the user&#39;s email address itself for this user. 
        /// </summary>
        public string AvatarUrl { get; private set; }

        /// <summary>
        /// A Zulip \&quot;client\&quot; string, describing what Zulip client sent the message. 
        /// </summary>
        public string Client { get; private set; }

        /// <summary>
        /// The content/body of the message. 
        /// </summary>
        public string Content { get; private set; }

        /// <summary>
        /// The HTTP &#x60;content_type&#x60; for the message content.  This will be &#x60;text/html&#x60; or &#x60;text/x-markdown&#x60;, depending on whether &#x60;apply_markdown&#x60; was set. 
        /// </summary>
        public string ContentType { get; private set; }

        /// <summary>
        /// Data on the recipient of the message; either the name of a stream or a dictionary containing basic data on the users who received the message. 
        /// </summary>
        public OneOfstringarray DisplayRecipient { get; private set; }

        /// <summary>
        /// The unique message ID.  Messages should always be displayed sorted by ID. 
        /// </summary>
        public int? Id { get; private set; }

        /// <summary>
        /// Whether the message is a [/me status message][status-messages]  [status-messages]: /help/format-your-message-using-markdown#status-messages 
        /// </summary>
        public bool? IsMeMessage { get; private set; }

        /// <summary>
        /// Data on any reactions to the message. 
        /// </summary>
        public List<EmojiReaction> Reactions { get; private set; }

        /// <summary>
        /// A unique ID for the set of users receiving the message (either a stream or group of users).  Useful primarily for hashing. 
        /// </summary>
        public int? RecipientId { get; private set; }

        /// <summary>
        /// The Zulip display email address of the message&#39;s sender. 
        /// </summary>
        public string SenderEmail { get; private set; }

        /// <summary>
        /// The full name of the message&#39;s sender. 
        /// </summary>
        public string SenderFullName { get; private set; }

        /// <summary>
        /// The user ID of the message&#39;s sender. 
        /// </summary>
        public int? SenderId { get; private set; }

        /// <summary>
        /// A string identifier for the realm the sender is in.  Unique only within the context of a given Zulip server.  E.g. on &#x60;example.zulip.com&#x60;, this will be &#x60;example&#x60;. 
        /// </summary>
        public string SenderRealmStr { get; private set; }

        /// <summary>
        /// Only present for stream messages; the ID of the stream. 
        /// </summary>
        public int? StreamId { get; private set; }

        /// <summary>
        /// The &#x60;topic&#x60; of the message.  Currently always &#x60;\&quot;\&quot;&#x60; for private messages, though this could change if Zulip adds support for topics in private message conversations.  The field name is a legacy holdover from when topics were called \&quot;subjects\&quot; and will eventually change. 
        /// </summary>
        public string Subject { get; private set; }

        /// <summary>
        /// Data on any links to be included in the &#x60;topic&#x60; line (these are generated by [custom linkification filters](/help/add-a-custom-linkifier) that match content in the message&#39;s topic.)  **Changes**: This field contained a list of urls before   Zulip 4.0 (feature level 46).  New in Zulip 3.0 (feature level 1): Previously, this field was called &#x60;subject_links&#x60;; clients are recommended to rename &#x60;subject_links&#x60; to &#x60;topic_links&#x60; if present for compatibility with older Zulip servers. 
        /// </summary>
        public List<MessagesBaseTopicLinks> TopicLinks { get; private set; }

        /// <summary>
        /// Data used for certain experimental Zulip integrations. 
        /// </summary>
        public List<string> Submessages { get; private set; }

        /// <summary>
        /// The UNIX timestamp for when the message was sent, in UTC seconds. 
        /// </summary>
        public int? Timestamp { get; private set; }

        /// <summary>
        /// The type of the message: &#x60;stream&#x60; or &#x60;private&#x60;. 
        /// </summary>
        public string Type { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use MessagesBase.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public MessagesBase()
        {
        }

        private MessagesBase(string AvatarUrl, string Client, string Content, string ContentType, OneOfstringarray DisplayRecipient, int? Id, bool? IsMeMessage, List<EmojiReaction> Reactions, int? RecipientId, string SenderEmail, string SenderFullName, int? SenderId, string SenderRealmStr, int? StreamId, string Subject, List<MessagesBaseTopicLinks> TopicLinks, List<string> Submessages, int? Timestamp, string Type)
        {
            
            this.AvatarUrl = AvatarUrl;
            
            this.Client = Client;
            
            this.Content = Content;
            
            this.ContentType = ContentType;
            
            this.DisplayRecipient = DisplayRecipient;
            
            this.Id = Id;
            
            this.IsMeMessage = IsMeMessage;
            
            this.Reactions = Reactions;
            
            this.RecipientId = RecipientId;
            
            this.SenderEmail = SenderEmail;
            
            this.SenderFullName = SenderFullName;
            
            this.SenderId = SenderId;
            
            this.SenderRealmStr = SenderRealmStr;
            
            this.StreamId = StreamId;
            
            this.Subject = Subject;
            
            this.TopicLinks = TopicLinks;
            
            this.Submessages = Submessages;
            
            this.Timestamp = Timestamp;
            
            this.Type = Type;
            
        }

        /// <summary>
        /// Returns builder of MessagesBase.
        /// </summary>
        /// <returns>MessagesBaseBuilder</returns>
        public static MessagesBaseBuilder Builder()
        {
            return new MessagesBaseBuilder();
        }

        /// <summary>
        /// Returns MessagesBaseBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>MessagesBaseBuilder</returns>
        public MessagesBaseBuilder With()
        {
            return Builder()
                .AvatarUrl(AvatarUrl)
                .Client(Client)
                .Content(Content)
                .ContentType(ContentType)
                .DisplayRecipient(DisplayRecipient)
                .Id(Id)
                .IsMeMessage(IsMeMessage)
                .Reactions(Reactions)
                .RecipientId(RecipientId)
                .SenderEmail(SenderEmail)
                .SenderFullName(SenderFullName)
                .SenderId(SenderId)
                .SenderRealmStr(SenderRealmStr)
                .StreamId(StreamId)
                .Subject(Subject)
                .TopicLinks(TopicLinks)
                .Submessages(Submessages)
                .Timestamp(Timestamp)
                .Type(Type);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(MessagesBase other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (MessagesBase.
        /// </summary>
        /// <param name="left">Compared (MessagesBase</param>
        /// <param name="right">Compared (MessagesBase</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (MessagesBase left, MessagesBase right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (MessagesBase.
        /// </summary>
        /// <param name="left">Compared (MessagesBase</param>
        /// <param name="right">Compared (MessagesBase</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (MessagesBase left, MessagesBase right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of MessagesBase.
        /// </summary>
        public sealed class MessagesBaseBuilder
        {
            private string _AvatarUrl;
            private string _Client;
            private string _Content;
            private string _ContentType;
            private OneOfstringarray _DisplayRecipient;
            private int? _Id;
            private bool? _IsMeMessage;
            private List<EmojiReaction> _Reactions;
            private int? _RecipientId;
            private string _SenderEmail;
            private string _SenderFullName;
            private int? _SenderId;
            private string _SenderRealmStr;
            private int? _StreamId;
            private string _Subject;
            private List<MessagesBaseTopicLinks> _TopicLinks;
            private List<string> _Submessages;
            private int? _Timestamp;
            private string _Type;

            internal MessagesBaseBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for MessagesBase.AvatarUrl property.
            /// </summary>
            /// <param name="value">The URL of the user&#39;s avatar.  Can be null only if client_gravatar was passed, which means that the user has not uploaded an avatar in Zulip, and the client should compute the gravatar URL by hashing the user&#39;s email address itself for this user. </param>
            public MessagesBaseBuilder AvatarUrl(string value)
            {
                _AvatarUrl = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Client property.
            /// </summary>
            /// <param name="value">A Zulip \&quot;client\&quot; string, describing what Zulip client sent the message. </param>
            public MessagesBaseBuilder Client(string value)
            {
                _Client = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Content property.
            /// </summary>
            /// <param name="value">The content/body of the message. </param>
            public MessagesBaseBuilder Content(string value)
            {
                _Content = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.ContentType property.
            /// </summary>
            /// <param name="value">The HTTP &#x60;content_type&#x60; for the message content.  This will be &#x60;text/html&#x60; or &#x60;text/x-markdown&#x60;, depending on whether &#x60;apply_markdown&#x60; was set. </param>
            public MessagesBaseBuilder ContentType(string value)
            {
                _ContentType = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.DisplayRecipient property.
            /// </summary>
            /// <param name="value">Data on the recipient of the message; either the name of a stream or a dictionary containing basic data on the users who received the message. </param>
            public MessagesBaseBuilder DisplayRecipient(OneOfstringarray value)
            {
                _DisplayRecipient = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Id property.
            /// </summary>
            /// <param name="value">The unique message ID.  Messages should always be displayed sorted by ID. </param>
            public MessagesBaseBuilder Id(int? value)
            {
                _Id = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.IsMeMessage property.
            /// </summary>
            /// <param name="value">Whether the message is a [/me status message][status-messages]  [status-messages]: /help/format-your-message-using-markdown#status-messages </param>
            public MessagesBaseBuilder IsMeMessage(bool? value)
            {
                _IsMeMessage = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Reactions property.
            /// </summary>
            /// <param name="value">Data on any reactions to the message. </param>
            public MessagesBaseBuilder Reactions(List<EmojiReaction> value)
            {
                _Reactions = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.RecipientId property.
            /// </summary>
            /// <param name="value">A unique ID for the set of users receiving the message (either a stream or group of users).  Useful primarily for hashing. </param>
            public MessagesBaseBuilder RecipientId(int? value)
            {
                _RecipientId = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.SenderEmail property.
            /// </summary>
            /// <param name="value">The Zulip display email address of the message&#39;s sender. </param>
            public MessagesBaseBuilder SenderEmail(string value)
            {
                _SenderEmail = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.SenderFullName property.
            /// </summary>
            /// <param name="value">The full name of the message&#39;s sender. </param>
            public MessagesBaseBuilder SenderFullName(string value)
            {
                _SenderFullName = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.SenderId property.
            /// </summary>
            /// <param name="value">The user ID of the message&#39;s sender. </param>
            public MessagesBaseBuilder SenderId(int? value)
            {
                _SenderId = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.SenderRealmStr property.
            /// </summary>
            /// <param name="value">A string identifier for the realm the sender is in.  Unique only within the context of a given Zulip server.  E.g. on &#x60;example.zulip.com&#x60;, this will be &#x60;example&#x60;. </param>
            public MessagesBaseBuilder SenderRealmStr(string value)
            {
                _SenderRealmStr = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.StreamId property.
            /// </summary>
            /// <param name="value">Only present for stream messages; the ID of the stream. </param>
            public MessagesBaseBuilder StreamId(int? value)
            {
                _StreamId = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Subject property.
            /// </summary>
            /// <param name="value">The &#x60;topic&#x60; of the message.  Currently always &#x60;\&quot;\&quot;&#x60; for private messages, though this could change if Zulip adds support for topics in private message conversations.  The field name is a legacy holdover from when topics were called \&quot;subjects\&quot; and will eventually change. </param>
            public MessagesBaseBuilder Subject(string value)
            {
                _Subject = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.TopicLinks property.
            /// </summary>
            /// <param name="value">Data on any links to be included in the &#x60;topic&#x60; line (these are generated by [custom linkification filters](/help/add-a-custom-linkifier) that match content in the message&#39;s topic.)  **Changes**: This field contained a list of urls before   Zulip 4.0 (feature level 46).  New in Zulip 3.0 (feature level 1): Previously, this field was called &#x60;subject_links&#x60;; clients are recommended to rename &#x60;subject_links&#x60; to &#x60;topic_links&#x60; if present for compatibility with older Zulip servers. </param>
            public MessagesBaseBuilder TopicLinks(List<MessagesBaseTopicLinks> value)
            {
                _TopicLinks = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Submessages property.
            /// </summary>
            /// <param name="value">Data used for certain experimental Zulip integrations. </param>
            public MessagesBaseBuilder Submessages(List<string> value)
            {
                _Submessages = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Timestamp property.
            /// </summary>
            /// <param name="value">The UNIX timestamp for when the message was sent, in UTC seconds. </param>
            public MessagesBaseBuilder Timestamp(int? value)
            {
                _Timestamp = value;
                return this;
            }

            /// <summary>
            /// Sets value for MessagesBase.Type property.
            /// </summary>
            /// <param name="value">The type of the message: &#x60;stream&#x60; or &#x60;private&#x60;. </param>
            public MessagesBaseBuilder Type(string value)
            {
                _Type = value;
                return this;
            }


            /// <summary>
            /// Builds instance of MessagesBase.
            /// </summary>
            /// <returns>MessagesBase</returns>
            public MessagesBase Build()
            {
                Validate();
                return new MessagesBase(
                    AvatarUrl: _AvatarUrl,
                    Client: _Client,
                    Content: _Content,
                    ContentType: _ContentType,
                    DisplayRecipient: _DisplayRecipient,
                    Id: _Id,
                    IsMeMessage: _IsMeMessage,
                    Reactions: _Reactions,
                    RecipientId: _RecipientId,
                    SenderEmail: _SenderEmail,
                    SenderFullName: _SenderFullName,
                    SenderId: _SenderId,
                    SenderRealmStr: _SenderRealmStr,
                    StreamId: _StreamId,
                    Subject: _Subject,
                    TopicLinks: _TopicLinks,
                    Submessages: _Submessages,
                    Timestamp: _Timestamp,
                    Type: _Type
                );
            }

            private void Validate()
            { 
            }
        }

        
    }
}
